cmd/common/utils/parser-tableformatter.go

```diff
@@ -97,3 +103,27 @@ func (p *GadgetParser[T]) BuildColumnsHeader() string {
 func (p *GadgetParser[T]) TransformToColumns(entry *T) string {
 	return p.formatter.FormatEntry(entry)
 }
+
+func (p *GadgetParser[T]) TransformToTable(entries []*T) string {
+	maxWidth := 0
+	if term.IsTerminal(int(os.Stdout.Fd())) {
+		terminalWidth, _, err := term.GetSize(0)
+		if err == nil {
+			maxWidth = terminalWidth
+		}
+	}
+
+	// Disable auto-scaling as AdjustWidthsToContent will already manage the
+	// screen size with the maxWidth we are passing.
+	// p.formatter.SetAutoScaling(false)
```

Comment:
Given the current structure of the code, I think it is necessary to have this helper. Otherwise, each gadget that is creating the parser needs to specify when calling `NewGadgetParser` whether it will print the full table or entry-by-entry so that we initialize the formatter with or without autoscaling. I would like to avoid that extra code, keep hiding such distinction from the gadget and manage all this in the parser.

Reply:
Solve using https://github.com/kinvolk/inspektor-gadget/pull/993.